src/pudl/etl/glue_assets.py

```diff
@@ -343,6 +344,28 @@ def epacamd_eia_subplant_ids(
             "when none are expected. Duplicates found: \n"
             f"{subplant_ids_updated[epacamd_eia_dupe_mask]}"
         )
+
+    # the network x process uses unit_id_pudl and generator_id. During processing,
+    # the boiler_id data are truncated and we only retain unique values for generator_id
+    # and unit_id_pudl. This step adds the lost boiler_id info back into the table.
+    subplant_ids_updated = pd.merge(
+        subplant_ids_updated.drop(columns="boiler_id").assign(
+            unit_id_pudl=lambda x: x.unit_id_pudl.astype(
+                "float"
+            )  # necessary step for tests
+        ),
+        boiler_generator_assn_eia860[
+            [
+                "plant_id_eia",
+                "generator_id",
+                "boiler_id",
+                "unit_id_pudl",
+            ]
+        ].drop_duplicates(),
+        on=["plant_id_eia", "generator_id", "unit_id_pudl"],
+        how="outer",
+    ).drop_duplicates()
+
```

Comment:
The reason that I added this chunk of code was because the network x process was dropping some of the boiler_ids (see comments above). This isn't pretty, but it remerges the subplant id table with the bga table so that we retain all the boiler-generator relationships. 

There are two main reasons it's not ideal: 

1) we already have a function called `augement_crosswalk_with_bga_eia860()` so it's weird / duplicative that we are merging the bga table with the subplant id table for a second time. (Can't find a way around this for now without changing how the network X is working which we agreed wasn't a good idea for right now). 

2) The `pytest test/unit/transform/glue_test.py::test_epacamd_eia_subplant_ids` was failing without the line that converts the `unit_id_pudl` field to a float. This step makes the tests pass, causes the asset build to fail -.- grr. I don't know how to get around it other than adding this column dtype conversion step into the tests rather than here. But that also feels janky. I would prefer to avoid this dtype issue if possible! It wasn't an issue before this merge was added which I don't really understand.

Reply:
on your second point, my first instinct is to add a `convert_dtypes()` step in the unit tests for all input dfs. hard agree that this convert to float is not ideal.

Reply:
and on #1... maybe we should slightly generalize the `augment_crosswalk_with_bga_eia860`. it looks like i? or you? idk added `boiler_id` into that merge. but if the nextwork_x step doesn't care about boilers/effectively drops boiler detail then they shouldn't be in the table pre-network_x anyway.

We could give `augment_crosswalk_with_bga_eia860` an `idx: list[str]` arg that is either just `["plant_id_eia", "generator_id"]` or `["plant_id_eia", "generator_id", "boiler_id"]` and always `boiler_generator_assn_eia860[idx + ["unit_id_pudl"]].drop_duplicates()`

Reply:
I don't think this function is used more than once so we could also just call it `augment_crosswalk_with_unit_id_pudl` and remove `boiler_id` from the merge cols